src/Build/Evaluation/Conditionals/StringExpressionNode.cs

```diff
@@ -98,6 +126,25 @@ internal override bool EvaluatesToEmpty(ConditionEvaluator.IConditionEvaluationS
             {
                 if (_expandable)
                 {
+                    if (_value.Length > 2)
+                    {
+                        if (_value[1] != '(' || _value[_value.Length - 1] != ')' || (_value[0] != '$' && _value[0] != '%' && _value[0] != '@'))
+                        {
+                            // This isn't just a property, item, or metadata value, and it isn't empty.
+                            return false;
+                        }
+                    }
+                    else if (_value.Length == 0)
+                    {
+                        _cachedExpandedValue = String.Empty;
+                        return true;
+                    }
+                    else
+                    {
+                        _cachedExpandedValue = _value;
+                        return false;
+                    }
```

Comment:
nit: Maybe use a `switch`? And/or comment, the last `else` block is especially hard to understand.

```suggestion
                    switch (_value.Length)
                    {
                        case 0: _cachedExpandedValue = String.Empty; return true;
                        case 1:
                        case 2: _cachedExpandedValue = _value; return false;
                        default:
                        {
                            if (_value[1] != '(' || _value[_value.Length - 1] != ')' || (_value[0] != '$' && _value[0] != '%' && _value[0] != '@'))
                            {
                                // This isn't just a property, item, or metadata value, and it isn't empty.
                                return false;
                            }                        
                        }
                    }
```

Reply:
Sounds good! The 1/2 case is for if it's nonempty but not long enough to have @/$/%, (, and ), so it has some characters that stay as they are. I can add a comment to make that clearer.